Csp compatible

lakshit-dua opened the pull request.

lakshit-dua committed:
- Make tiny-slider CSP compliant

lakshit-dua committed:
- Added helper functions
- Create stylesheet now adopts the stylesheet for shadow roots
- Multiple css rules handling
- Added minification throught rollup, now creates tiny-slider.min.js
- Code cleanup

lakshit-dua committed:
- Changed mentions of ion

Chirag-Khanna commented on src/helpers/addCSSRule.js:
```diff
@@ -2,8 +2,19 @@
-}
\ No newline at end of file
+    'replaceSync' in sheet ?
+    sheet.replaceSync(sheet.cssRules.length > 0 ? getCompleteRules(sheet, selector + '{' + rules + '}') : selector + '{' + rules + '}') :
```
change the name for getCompleteRules

Chirag-Khanna commented on src/helpers/addCSSRule.js:
```diff
@@ -2,8 +2,19 @@
+function getCompleteRules(sheet, cssText) {
+  let completeCSSText = "";
+  let i;
+  for (i = 0; i < sheet.cssRules.length; i++) {
```
move let inside for, use right variable name

Chirag-Khanna commented on src/helpers/checkConstructStyleSheet.js:
```diff
@@ -0,0 +1,12 @@
+export function checkConstructStyleSheet() {
```
check if we can run this fn only ones

Chirag-Khanna commented on src/helpers/createStyleSheet.js:
```diff
@@ -1,26 +1,19 @@
+      let shadowRoot = container.getRootNode();
+      shadowRoot.adoptedStyleSheets = shadowRoot.adoptedStyleSheets ? [...shadowRoot.adoptedStyleSheets, sheet] : [sheet]; 
+    } else {
+      document.adoptedStyleSheets = document.adoptedStyleSheets ? [...document.adoptedStyleSheets, sheet] : [sheet];
```
always adopt stylesheet in document even if being added in shadow dom

Chirag-Khanna commented on src/helpers/customCssStyleSheet.js:
```diff
@@ -0,0 +1,31 @@
+import { forEach } from "./forEach";
+import { getCamelCase } from "./getCamelCase";
+export function CustomCssStyleSheet() {
```
We should not have our properties in prototype. Let's move everything in a class

Chirag-Khanna commented on src/helpers/getCamelCase.js:
```diff
@@ -0,0 +1,24 @@
+    let camelCaseWords = [];
+    forEach(rules.split(';'), (rule) => {
+      let [ruleNameDashed, value] = rule.split(':');
+      let ruleWords = ruleNameDashed.split('-');
```
update string manipulation to make this better.

Chirag-Khanna commented on src/helpers/customCssStyleSheet.js:
```diff
@@ -0,0 +1,31 @@
+
+CustomCssStyleSheet.prototype.rules = {};
+CustomCssStyleSheet.prototype.rulesList = [];
+CustomCssStyleSheet.prototype.ruleLength = 0;
```
remove ruleLength

Chirag-Khanna commented on src/helpers/getCssRulesLength.js:
```diff
@@ -1,4 +1,3 @@
 export function getCssRulesLength(sheet) {
-  var rule = ('insertRule' in sheet) ? sheet.cssRules : sheet.rules;
-  return rule.length;
+  return ('replaceSync' in sheet) ? sheet.cssRules : sheet.ruleLength;
```
remove ruleLength here.

Chirag-Khanna commented on src/helpers/customCssStyleSheet.js:
```diff
@@ -0,0 +1,31 @@
+}
+
+CustomCssStyleSheet.prototype.rules = {};
+CustomCssStyleSheet.prototype.rulesList = [];
```
use name rules instead of rulesList

Chirag-Khanna commented on src/tiny-slider.js:
```diff
@@ -2059,6 +2061,29 @@ export var tns = function(options) {
   // set duration
   function resetDuration (el, str) {
     if (TRANSITIONDURATION) { el.style[TRANSITIONDURATION] = str; }
+    resetInlineStyles(el);
```
update handling for add and remove class

lakshit-dua committed:
- Code review points
- Now inline styles are added/removed whenever relevent elements' classes are altered

lakshit-dua committed:
- Removed duplicacy and not needed code

Chirag-Khanna commented on src/helpers/createStyleSheet.js:
```diff
@@ -1,26 +1,17 @@
-    container.getRootNode().appendChild(style);
+// create and append style sheet
+export function createStyleSheet (media, nonce, container, HASCONSTRUCTIBLESTYLESHEET) {
+  if (HASCONSTRUCTIBLESTYLESHEET) {
```
naming seems a problem for HASCONSTRUCTIBLESTYLESHEET

Chirag-Khanna commented on src/helpers/customCssStyleSheet.js:
```diff
@@ -0,0 +1,44 @@
+  }
+
+  addRule(selector, rules, index) {
+    this.rules[index] = { selector, rules };
```
if not index is passed. also if a overriding index is passed. what about the item that was already present there?

Chirag-Khanna commented on src/helpers/customCssStyleSheet.js:
```diff
@@ -0,0 +1,44 @@
+    this.rules[index] = { selector, rules };
+    this.rulesMap[selector] = rules;
+    if (rules.length > 0) {
+      let keyValuePairs = getCamelCase(rules);
```
I see we are still applying the rules when addRule is called

Chirag-Khanna commented on src/helpers/getCssRulesLength.js:
```diff
@@ -1,4 +1,3 @@
 export function getCssRulesLength(sheet) {
-  var rule = ('insertRule' in sheet) ? sheet.cssRules : sheet.rules;
-  return rule.length;
+  return ('replaceSync' in sheet) ? sheet.cssRules : (sheet.rules && sheet.rules.length);
```
we need length also for cssrules scenario

Chirag-Khanna commented on src/helpers/inlineStylesHandler.js:
```diff
@@ -0,0 +1,41 @@
+      return;
+    }
+    var doc;
+    if (el.tagName === 'SLOT') {
```
you should not check el.tagname and instead check if the el lies in a shadow dom then gets its shadow root

Chirag-Khanna commented:
Also we need to check in MAC all the scenarios. For the performance is also a concern

lakshit-dua committed:
- Code review points

lakshit-dua committed:
- More code review points
